core/livepeernode.go

```
@@ -78,24 +88,28 @@ func NewLivepeerNode(e eth.LivepeerEthClient, vn net.VideoNetwork, nodeId NodeID
 		return nil, ErrLivepeerNode
 	}
 
-	return &LivepeerNode{VideoCache: NewBasicVideoCache(vn), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd, PeerConns: make([]PeerConn, 0)}, nil
+	return &LivepeerNode{VideoCache: NewBasicVideoCache(vn), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd}, nil
 }
 
 //Start sets up the Livepeer protocol and connects the node to the network
-func (n *LivepeerNode) Start(ctx context.Context, bootID, bootAddr string) error {
+func (n *LivepeerNode) Start(ctx context.Context, bootIDs, bootAddrs []string) error {
+	if len(bootIDs) != len(bootAddrs) {
+		return errors.New("BootIDs and BootAddrs do not match.")
+	}
 	//Set up protocol (to handle incoming streams)
 	if err := n.VideoNetwork.SetupProtocol(); err != nil {
 		glog.Errorf("Error setting up protocol: %v", err)
 		return err
 	}
 
 	//Connect to bootstrap node.  This currently also kicks off a bootstrap process, which periodically checks for new peers and connect to them.
-	if bootID != "" && bootAddr != "" {
-		glog.Infof("Connecting to %v %v", bootID, bootAddr)
-		if err := n.VideoNetwork.Connect(bootID, []string{bootAddr}); err != nil {
-			glog.Errorf("Cannot connect to node: %v", err)
-		} else {
-			n.PeerConns = append(n.PeerConns, PeerConn{NodeID: bootID, NodeAddr: bootAddr})
+	if len(bootIDs) > 0 && len(bootAddrs) > 0 {
+		for i, bootID := range bootIDs {
```

If all the bootnode connection attempts fail should we return an error?